File: kt/moq/src/jvmAndAndroidMain/kotlin/dev/moq/Moq.kt
package dev.moq

import uniffi.moq.MoqClient
import uniffi.moq.MoqSession

/**
 * Top-level entry points for the Moq protocol stack.
 */
object Moq {
    /** Connect to a MoQ relay using default client configuration. */
    suspend fun connect(url: String): MoqSession = MoqClient().use { it.connect(url) }

    /** Build a client with custom configuration before connecting. */
    fun client(): MoqClient = MoqClient()
}

File: swift/Sources/Moq/Moq.swift
import Foundation
import MoqFFI

/// Top-level entry points for the Moq protocol stack.
public enum Moq {
    /// Connect to a MoQ relay using default client configuration.
    public static func connect(url: String) async throws -> MoqSession {
        let client = MoqClient()
        return try await client.connect(url: url)
    }

    /// Build a client with custom configuration before connecting.
    public static func client() -> MoqClient {
        MoqClient()
    }
}
